Add PR CI workflow for lint and build

Create a GitHub Actions workflow that runs on pull requests and pushes, executing 
> polycule-graph@0.3.0 lint /workspace/polycule-graph
> eslint .


/workspace/polycule-graph/src/components/GraphView.tsx
  75:6  warning  React Hook useEffect has missing dependencies: 'nodes.length' and 'triggerLayout'. Either include them or remove the dependency array  react-hooks/exhaustive-deps

✖ 1 problem (0 errors, 1 warning) and 
> polycule-graph@0.3.0 build /workspace/polycule-graph
> tsc -b && vite build

vite v7.0.4 building for production...
transforming...
✓ 1741 modules transformed.
rendering chunks...
computing gzip size...
dist/index.html                   0.47 kB │ gzip:   0.31 kB
dist/assets/index-B8zGM29U.css   29.24 kB │ gzip:   6.27 kB
dist/assets/index-AyizBLNg.js   807.39 kB │ gzip: 262.15 kB
✓ built in 2.76s (typecheck + Vite build). Keep it separate from the GitHub Pages deploy workflow.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -1,8 +1,8 @@
 name: CI
 
 on:
-  push:
   pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
 
 permissions:
   contents: read

